### ISSUE TITLE
keras tuner for keras functional models

### ISSUE THREAD

**Issue by amirzlq5** [keras-team/keras-tuner#281]
Is it possible to use keras tuner in keras functional models?

**Comment by hassancpu** [comment]
I have the same question...

**Comment by amirzlq5** [comment]
After studying, I came to the conclusion that the Keras tuner is not suitable for models with multiple inputs (functional models).

**Comment by Jimmy2027** [comment]
Would also interest me

**Comment by hassancpu** [comment]
> After studying, I came to the conclusion that the Keras tuner is not suitable for models with multiple inputs (functional models).

What is the appropriate tuner for such models?

**Comment by amirzlq5** [comment]
The only hyperparameter optimization library I've found that works with Keras functional API is Talos.

https://github.com/autonomio/talos/blob/9890d71d31451af3d7e8d91a75841bc7904db958/docs/Examples_Multiple_Inputs.md

**Comment by amirzlq5** [comment]
> 
> 
> > After studying, I came to the conclusion that the Keras tuner is not suitable for models with multiple inputs (functional models).
> 
> What is the appropriate tuner for such models?

The only hyperparameter optimization library I've found that works with Keras functional API is Talos.

https://github.com/autonomio/talos/blob/9890d71d31451af3d7e8d91a75841bc7904db958/docs/Examples_Multiple_Inputs.md

**Comment by Jimmy2027** [comment]
actually they have an example where they use keras functional API: https://github.com/keras-team/keras-tuner/blob/master/examples/cifar10.py

**Comment by omalleyt12** [comment]
@amirzlq5 Thanks for the issue!

Yes it's possible to use Keras Tuner with any Keras model type, please see the linked example in the comment above

**Comment by Sanjaykrishnamurthy** [comment]
@Jimmy2027  and omalleyt12,

I could see only one Input layer in the example. Am I missing something? 
My use-case requires multiple Input layers of different dimensions.

**Comment by Sanjaykrishnamurthy** [comment]
> > > After studying, I came to the conclusion that the Keras tuner is not suitable for models with multiple inputs (functional models).
> > 
> > 
> > What is the appropriate tuner for such models?
> 
> The only hyperparameter optimization library I've found that works with Keras functional API is Talos.
> 
> https://github.com/autonomio/talos/blob/9890d71d31451af3d7e8d91a75841bc7904db958/docs/Examples_Multiple_Inputs.md

Even this works for single input models

**Comment by JamesCallanan** [comment]
For anyone looking for the example referenced above, check either of the following
- https://github.com/spellml/examples/blob/master/keras/cifar10_cnn.py
- https://spell.ml/blog/an-introduction-to-hyperparameter-search-with-cifar10-Xo8_6BMAACEAkwVs

**Comment by JamesCallanan** [comment]
Example can be seen here

https://wandb.ai/arig23498/keras-tuner/reports/Keras-Tuner-with-W-B--Vmlldzo0MzQ1NzU

def build_model(hp):
    """
    Builds a convolutional model.
    
    Args:
      hp: Hyperparamet object, This is the object that helps
        us sample hyperparameter for a particular trial.
    
    Returns:
      model: Keras model, Returns a keras model.
    """
    inputs = tf.keras.Input(shape=(28, 28, 1))
    x = inputs
    # In this example we also get to look at
    # conditional heyperparameter settings.
    # Here the `kernel_size` is conditioned
    # with the for loop counter. 
    for i in range(hp.Int('conv_layers', 1, 3)):
      x = tf.keras.layers.Conv2D(
          filters=hp.Int('filters_' + str(i), 4, 32, step=4, default=8),
          kernel_size=hp.Int('kernel_size_' + str(i), 3, 5),
          activation='relu',
          padding='same')(x)
      # choosing between max pool and avg pool
      if hp.Choice('pooling' + str(i), ['max', 'avg']) == 'max':
        x = tf.keras.layers.MaxPooling2D()(x)
      else:
        x = tf.keras.layers.AveragePooling2D()(x)
      x = tf.keras.layers.BatchNormalization()(x)
      x = tf.keras.layers.ReLU()(x)

    if hp.Choice('global_pooling', ['max', 'avg']) == 'max':
      x = tf.keras.layers.GlobalMaxPooling2D()(x)
    else:
      x = tf.keras.layers.GlobalAveragePooling2D()(x)
    outputs = tf.keras.layers.Dense(10, activation='softmax')(x)
    
    model = tf.keras.Model(inputs, outputs)
    return model

**Comment by fesaaved** [comment]
> @amirzlq5 Thanks for the issue!
> 
> Yes it's possible to use Keras Tuner with any Keras model type, please see the linked example in the comment above

Dear @omalleyt12, 
This link doesn't work, unfortunately: https://github.com/keras-team/keras-tuner/blob/master/examples/cifar10.py
I don't see a solution to the original problem in the other examples provided. So if the model's input is irregular, let's say input=[x_train_2d, x_train_1d], none of the examples solve the issue. Am I missing something?

Best regards
